### PR TITLE
Support parent fields in getfield and putfield

### PR DESCRIPTION
**Pull request by hankluo6**
When pitifulvm create a new object, it will create the object itself and all its parents as objects and use linked list to link all related objects. When `getfield` or `putfield` are called, pitifulvm will try to find fields along the linked list.

Update `Inherit.java` in order to test parent fields.

### CODE REVIEW

File: classfile.h [hunk 1]
```diff
@@ -54,13 +54,15 @@ typedef struct {
...
 
-typedef struct {
+typedef struct class_file {
     constant_pool_t constant_pool;
     class_info_t *info;
     method_t *methods;
     field_t *fields;
     u2 fields_count;
     bool initialized;
+    struct class_file *next;
```
Comment (jserv) [inline classfile.h:64]: Would you introduce some macros / inline functions as linked list helpers?

Comment (hankluo6) [reply classfile.h:64]: ok I will add some macros.

File: jvm.c [hunk 2]
```diff
@@ -1197,22 +1206,21 @@ stack_entry_t *execute(method_t *method,
...
-                if (!strcmp(class_name, "java/lang/Object"))
-                    break;
             }
 
-            /* call static initialization */
-            while (count) {
-                target_class = stack[--count];
+            /* reversely call static initialization if class have not been
+             * initialized */
+            list_for_each(target_class, list)
```
Comment (jserv) [inline jvm.c]: Update `.clang-format` to support this macro:
```
ForEachMacros:
  - list_for_each
```

Then, `clang-format` can indent better.

File: list.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#define init_list(list) list->prev = list->next = list;
```
Comment (jserv) [inline list.h:3]: Add `#pragma once` at the beginning of this header.

**Pull request merged**